tests/gbs/test_similarity.py

```diff
@@ -194,3 +196,94 @@ def test_sample_max_count_deterministic(self, photon_num, count, monkeypatch):
             m.setattr("numpy.random.choice", lambda x: x[0])
             samp = similarity.event_to_sample(photon_num, count, modes_dim)
         assert samp[0] == count
+
+
+def test_orbit_cardinality():
+    """Test if function ``strawberryfields.gbs.similarity.orbit_cardinality`` returns the
+    correct number of samples for some hard-coded examples."""
+
+    orbits = {
+        ((1, 1, 2), 4): 12,
+        ((1, 1), 4): 6,
+        ((1, 2, 3), 4): 24,
+        ((1, 1, 1, 1), 5): 5,
+        ((1, 1, 2), 5): 30,
+        ((1, 2, 3), 5): 60,
+    }
+
+    calc_cardinalities = {}
+    for o, _ in orbits.items():
+        calc_cardinalities[o] = similarity.orbit_cardinality(list(o[0]), o[1])
+
+    assert calc_cardinalities == orbits
```

Comment:
This is a minor design comment, but you could parametrize this test like so:
```python
@pytest.mark.parametrize("orbit, max_photon, expected", orbits.items())
def test_orbit_cardinality(orbit, max_photon, expected):
```
with very minimal changes. This would just lead to slightly nicer test output --- the failing case would be much more easily identified.